### ISSUE TITLE
panic when closing tfchaind with CTRL+D

### ISSUE THREAD

**Issue by GlenDC** [threefoldfoundation/tfchain#82]
```
Caught stop signal, quitting...
Closing block creator...
Closing wallet...
Closing transaction pool...
Closing explorer...
Closing consensus set...
goroutine 2481 [running]:
runtime/debug.Stack(0x1224887, 0xc4208c6ea8, 0xc4208c6e70)
        /usr/local/go/src/runtime/debug/stack.go:24 +0xb5
runtime/debug.PrintStack()
        /usr/local/go/src/runtime/debug/stack.go:16 +0x34
github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/build.Critical(0xc4208c6f58, 0x2, 0x2)
        $HOME/go/src/github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/build/critical.go:15 +0xa7
github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/explorer.(*Explorer).ProcessConsensusChange(0xc42021cd80, 0x50d2436a6c4fc514, 0xcc17039e90b0404b, 0x185a25ff612168ad, 0x2557f556bcd16694, 0x0, 0x0, 0x0, 0xc4200d7180, 0x1, ...)
        $HOME/go/src/github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/explorer/update.go:199 +0x193
github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/consensus.(*ConsensusSet).readlockUpdateSubscribers(0xc42004ec00, 0x0, 0x0, 0x0, 0xc420cfa040, 0x1, 0x1)
        $HOME/go/src/github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/consensus/subscribe.go:85 +0x23c
github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/consensus.(*ConsensusSet).managedAcceptBlock(0xc42004ec00, 0x3ff089d5c5e88e10, 0xa10ae2ce36c08dbd, 0xaf163408c0c60892, 0xb7f063ab8f8cc7e1, 0x5aebee1e, 0x274d, 0x0, 0x0, 0xc420478460, ...)
        $HOME/go/src/github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/consensus/accept.go:268 +0x2e0
github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/consensus.(*ConsensusSet).managedReceiveBlocks(0xc42004ec00, 0x1852da0, 0xc4202e5300, 0x0, 0x0)
        $HOME/go/src/github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/consensus/synchronize.go:220 +0x492
github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/consensus.(*ConsensusSet).(github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/consensus.managedReceiveBlocks)-fm(0x1852da0, 0xc4202e5300, 0x0,0x0)
        $HOME/go/src/github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/consensus/synchronize.go:397 +0x56
github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/gateway.(*Gateway).managedRPC(0xc42029e000, 0xc4200296c0, 0x13, 0x17c6f77, 0xa, 0xc4201ec040, 0x0, 0x0)
        $HOME/go/src/github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/gateway/rpc.go:66 +0x35e
github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/gateway.(*Gateway).RPC(0xc42029e000, 0xc4200296c0, 0x13, 0x17c6f77, 0xa, 0xc4201ec040, 0x0, 0x0)
        $HOME/go/src/github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/gateway/rpc.go:76 +0xd2
github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/consensus.(*ConsensusSet).threadedRPCRelayHeader.func3(0xc42004ec00, 0x1852da0, 0xc4204fc020, 0xc4202fa050)
        $HOME/go/src/github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/consensus/synchronize.go:397 +0x126
created by github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/consensus.(*ConsensusSet).threadedRPCRelayHeader
        $HOME/go/src/github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/consensus/synchronize.go:396 +0x5c4
Critical error: explorer update failed: database not open
Please submit a bug report here: https://github.com/rivine/rivine/issues
panic: Critical error: explorer update failed: database not open
Please submit a bug report here: https://github.com/rivine/rivine/issues


goroutine 2481 [running]:
github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/build.Critical(0xc4208c6f58, 0x2, 0x2)
        $HOME/go/src/github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/build/critical.go:18 +0x121
github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/explorer.(*Explorer).ProcessConsensusChange(0xc42021cd80, 0x50d2436a6c4fc514, 0xcc17039e90b0404b, 0x185a25ff612168ad, 0x2557f556bcd16694, 0x0, 0x0, 0x0, 0xc4200d7180, 0x1, ...)
        $HOME/go/src/github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/explorer/update.go:199 +0x193
github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/consensus.(*ConsensusSet).readlockUpdateSubscribers(0xc42004ec00, 0x0, 0x0, 0x0, 0xc420cfa040, 0x1, 0x1)
        $HOME/go/src/github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/consensus/subscribe.go:85 +0x23c
github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/consensus.(*ConsensusSet).managedAcceptBlock(0xc42004ec00, 0x3ff089d5c5e88e10, 0xa10ae2ce36c08dbd, 0xaf163408c0c60892, 0xb7f063ab8f8cc7e1, 0x5aebee1e, 0x274d, 0x0, 0x0, 0xc420478460, ...)
        $HOME/go/src/github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/consensus/accept.go:268 +0x2e0
github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/consensus.(*ConsensusSet).managedReceiveBlocks(0xc42004ec00, 0x1852da0, 0xc4202e5300, 0x0, 0x0)
        $HOME/go/src/github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/consensus/synchronize.go:220 +0x492
github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/consensus.(*ConsensusSet).(github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/consensus.managedReceiveBlocks)-fm(0x1852da0, 0xc4202e5300, 0x0,0x0)
        $HOME/go/src/github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/consensus/synchronize.go:397 +0x56
github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/gateway.(*Gateway).managedRPC(0xc42029e000, 0xc4200296c0, 0x13, 0x17c6f77, 0xa, 0xc4201ec040, 0x0, 0x0)
        $HOME/go/src/github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/gateway/rpc.go:66 +0x35e
github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/gateway.(*Gateway).RPC(0xc42029e000, 0xc4200296c0, 0x13, 0x17c6f77, 0xa, 0xc4201ec040, 0x0, 0x0)
        $HOME/go/src/github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/gateway/rpc.go:76 +0xd2
github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/consensus.(*ConsensusSet).threadedRPCRelayHeader.func3(0xc42004ec00, 0x1852da0, 0xc4204fc020, 0xc4202fa050)
        $HOME/go/src/github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/consensus/synchronize.go:397 +0x126
created by github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/consensus.(*ConsensusSet).threadedRPCRelayHeader
        $HOME/go/src/github.com/threefoldfoundation/tfchain/vendor/github.com/rivine/rivine/modules/consensus/synchronize.go:396 +0x5c4
```

**Comment by GlenDC** [comment]
duplicate of #64?

**Comment by LeeSmet** [comment]
Seems like a duplicate indeed. In the meantime I was able to reproduce the problem by having a node with the explorer module sync, and then sending ctrl+C while it was doing the IBD

**Comment by LeeSmet** [comment]
Fixed in rivine/rivine#391
